Avoid None as value for referenced_columns in qgsfunction.py docs

gacarrillor opened the pull request:
Use empty list instead.

When using None, as it is currently suggested by the docs from PR #52841 on, users will get this Python error:

```
An error has occurred while executing Python code: 

TypeError: invalid result from QgsPyExpressionFunction.referencedColumns(), NoneType cannot be converted to a C/C++ QSet in this context 

```

This adds consistency with the template used in the expression function editor, which uses an empty list.

(References #45519)

------------------
@YoannQDQ, are you fine with this minor change?

gacarrillor committed:
Avoid None as value for referenced_columns (use empty list instead) in qgsfunction.py documentation

YoannQDQ commented:
Sure 👍

m-kuhn commented on python/core/additions/qgsfunction.py:
```diff
@@ -118,7 +118,7 @@ def register_function(
     :param group: the expression group in which the function should be added
     :param usesgeometry: Defines if this expression requires the geometry. By default False.
-    :param referenced_columns: An array of field names on which this expression works. By default ``[QgsFeatureRequest.ALL_ATTRIBUTES]``. Can be set to None for slightly faster evaluation.
+    :param referenced_columns: An array of field names on which this expression works. By default ``[QgsFeatureRequest.ALL_ATTRIBUTES]``. Can be set to an empty list for slightly faster evaluation.
```
```suggestion
    :param referenced_columns: An array of names of fields on which this expression works. By default ``[QgsFeatureRequest.ALL_ATTRIBUTES]``. Specifying a subset of fields or an empty list will result in a faster execution.
```

I think this can make a big difference in some scenarios, so removing the "slightly" sounds appropriate. The rest is a proposal for a reformulation.

gacarrillor replied:
I agree, we could even add that such performance improvement depends on the provider, but that's probably too low-level-info for end users.

gacarrillor committed:
Update python/core/additions/qgsfunction.py

Avoid using 'slightly' since this could be a considerable performance improvement depending on the provider.

Co-authored-by: Matthias Kuhn <matthias@opengis.ch>

m-kuhn commented on python/core/additions/qgsfunction.py:
```diff
@@ -185,7 +185,7 @@ def qgsfunction(args="auto", group="custom", **kwargs):
           Defines if this expression requires the geometry. By default False.
         * *referenced_columns* (``list``) --
-          An array of field names on which this expression works. By default ``[QgsFeatureRequest.ALL_ATTRIBUTES]``. Can be set to None for slightly faster evaluation.
+          An array of field names on which this expression works. By default ``[QgsFeatureRequest.ALL_ATTRIBUTES]``. Can be set to an empty list for slightly faster evaluation.
```
@gacarrillor should the two texts be the same?

gacarrillor replied:
Done :+1:

gacarrillor committed:
Update qgsfunction.py, homogenize referenced_columns subsection
